lib/experimental/new_embed.dart

```diff
@@ -43,6 +43,38 @@ class NewEmbedOptions {
   NewEmbedOptions(this.mode);
 }
 
+enum DialogResult {
+  yes,
+  no,
+}
+
+Future<DialogResult> showDialog(String title, String htmlMessage,
+    {String yesText = "Yes", String noText = "No"}) {
+  querySelector('#my-dialog-title').text = title;
+  querySelector('#my-dialog-content')
+      .setInnerHtml(htmlMessage, validator: PermissiveNodeValidator());
+
+  var yesButton = querySelector('#dialog-yes')..text = yesText;
+  var noButton = querySelector('#dialog-no')..text = noText;
+  var dialog = MDCDialog(document.querySelector('.mdc-dialog'));
```

Comment:
The paradigm we use elsewhere in the file is that long-lived parts of the DOM get a long-lived Dart object to represent them. We have one TabController object that lives forever, for example, since the tab HTML is in the original file, but the HTML that's created on the fly for individual analyzer messages doesn't get that kind of treatment.

If I'm reading this right, `showDialog` would be a departure from that. Would it be better to create a long-lived dialog object that's repeatedly opened and closed, or is this definitely the way to go?

Reply:
Also, do we care what happens if the dialog's already open when `showDialog` is called?

Reply:
Also: this change helps me, as I need a dialog for #1065. 😄

Reply:
I think it's a good point - it's probably better to keep the MDCDialog object stored so that we aren't repeatedly telling the library to create a new component.

The problem is that I imagine this MDCDialog object would be stored on NewEmbed, but this method is currently used by NewEmbedTabController. I don't think it makes sense to store it there...

Reply:
Also a good point, and the ability to pop up a dialog is likely to be a cross-cutting concern for other parts of the UI.

We could make a global method to pop up the dialog, but that's ugly as well. What about injecting a dialog-launching function to NewEmbedTabController as one of its constructor params?

Reply:
Discussed offline and agreed on a slightly different approach. @johnpryan is going to make a Dialog class, an instance of which will be owned by NewEmbed and given to any sub-components that need to show a dialog.